kintoclient/__init__.py

```diff
@@ -0,0 +1,344 @@
+import requests
+import urlparse
+import json
+import six
+import uuid
+
+DEFAULT_SERVER_URL = 'https://kinto.dev.mozaws.net/v1'
+
+OBJECTS_PERMISSIONS = {
+    'bucket': ['group:create', 'collection:create', 'write', 'read'],
+    'group': ['write', 'read'],
+    'collection': ['write', 'read', 'record:create'],
+    'record': ['read', 'write']
+}
+
+ID_FIELD = 'id'
+
+
+class KintoException(Exception):
+    pass
+
+
+class BucketNotFound(KintoException):
+    pass
+
+
+def create_session(server_url=None, auth=None, session=None):
+    """Returns a session from the passed arguments.
+
+    :param server_url:
+        The URL of the server to use, with the prefix.
+    :param auth:
+        A requests authentication policy object.
+    :param session:
+        An optional session object to use, rather than creating a new one.
+    """
+    # XXX Refactor the create_session to take place in the caller objects.
+    # E.g. test if the session exists before calling create_session.
+    if session is not None and (
+            server_url is not None or auth is not None):
+        msg = ("You cannot specify session and server_url or auth. "
+               "Chose either session or auth + server_url.")
+        raise AttributeError(msg)
+    if session is None and server_url is None and auth is None:
+        msg = ("You need to either set session or auth + server_url")
+        raise AttributeError(msg)
+    if session is None:
+        session = Session(server_url=server_url, auth=auth)
+    return session
+
+
+class Session(object):
+    """Handles all the interactions with the network.
+    """
+    def __init__(self, server_url=DEFAULT_SERVER_URL, auth=None):
+        self.server_url = server_url
+        self.auth = auth
+
+    def request(self, method, url, data=None, permissions=None, **kwargs):
+        actual_url = urlparse.urljoin(self.server_url, url)
+        if self.auth is not None:
+            kwargs.setdefault('auth', self.auth)
+
+        payload = {}
+        # if data is not None:
+        payload['data'] = data or {}
+        if permissions is not None:
+            if hasattr(permissions, 'as_dict'):
+                permissions = permissions.as_dict()
+            payload['permissions'] = permissions
+        if payload:
+            kwargs.setdefault('headers', {})\
+                  .setdefault('Content-Type', 'application/json')
+            kwargs.setdefault('data', json.dumps(payload))
+        resp = requests.request(method, actual_url, **kwargs)
+        if not (200 <= resp.status_code < 400):
```

Comment:
it might be more straight forward to call and catch [resp.raise_for_status()](http://docs.python-requests.org/en/latest/api/#requests.Response.raise_for_status)